src/geo/projection/globe_tile_transform.js

```diff
@@ -0,0 +1,138 @@
+        const decode = globeDenormalizeECEF(globeTileBounds(id.canonical));
+        return mat4.multiply([], this._globeMatrix, decode);
+    }
+
+    createInversionMatrix(id: UnwrappedTileID): mat4 {
+        const center = this._tr.center;
+        const ecefUnitsToPixels = globeECEFUnitsToPixelScale(this._worldSize);
+        const matrix = mat4.identity(new Float64Array(16));
+        const encode = globeNormalizeECEF(globeTileBounds(id.canonical));
+        mat4.multiply(matrix, matrix, encode);
+        mat4.rotateY(matrix, matrix, degToRad(center.lng));
+        mat4.rotateX(matrix, matrix, degToRad(center.lat));
+
+        mat4.scale(matrix, matrix, [1.0 / ecefUnitsToPixels, 1.0 / ecefUnitsToPixels, 1.0]);
+
+        const PPMMercator = mercatorZfromAltitude(1.0, center.lat) * this._worldSize;
+        const globeToMercatorPPMRatio = PPMMercator / this._tr.pixelsPerMeter;
+        const worldSizeMercator = this._worldSize / globeToMercatorPPMRatio;
+        const wsRadius = worldSizeMercator / (2.0 * Math.PI);
+        const localRadius = EXTENT / (2.0 * Math.PI);
+        const ecefUnitsToMercatorPixels = wsRadius / localRadius;
+
+        const scaling = mat4.identity(new Float64Array(16));
+        mat4.scale(scaling, scaling, [ecefUnitsToMercatorPixels, ecefUnitsToMercatorPixels, 1.0]);
+
+        return mat4.multiply(matrix, matrix, scaling);
+    }
+
+    upVector(id: CanonicalTileID, x: number, y: number): vec3 {
+        const corners = globeTileLatLngCorners(id);
+        const tl = corners[0];
+        const br = corners[1];
+
+        const tlUp = latLngToECEF(tl[0], tl[1]);
+        const trUp = latLngToECEF(tl[0], br[1]);
+        const brUp = latLngToECEF(br[0], br[1]);
+        const blUp = latLngToECEF(br[0], tl[1]);
+
+        vec3.normalize(tlUp, tlUp);
+        vec3.normalize(trUp, trUp);
+        vec3.normalize(brUp, brUp);
+        vec3.normalize(blUp, blUp);
+
+        const u = x / EXTENT;
+        const v = y / EXTENT;
+
+        const tltr = vec3.lerp([], tlUp, trUp, u);
+        const blbr = vec3.lerp([], blUp, brUp, u);
+
+        return vec3.lerp([], tltr, blbr, v);
+    }
+
+    upVectorScale(id: CanonicalTileID): number {
+        const pixelsPerMeterECEF = mercatorZfromAltitude(1, 0.0) * 2.0 * GLOBE_RADIUS * Math.PI;
+        return pixelsPerMeterECEF * globeECEFNormalizationScale(globeTileBounds(id));
+    }
+
+    pointCoordinate(x: number, y: number): MercatorCoordinate {
+        const p0 = [x, y, 0, 1];
+        const p1 = [x, y, 1, 1];
+
+        vec4.transformMat4(p0, p0, this._tr.pixelMatrixInverse);
+        vec4.transformMat4(p1, p1, this._tr.pixelMatrixInverse);
+
+        vec4.scale(p0, p0, 1 / p0[3]);
+        vec4.scale(p1, p1, 1 / p1[3]);
+
+        const p0p1 = vec3.sub([], p1, p0);
+        const dir = vec3.normalize([], p0p1);
+
+        // Compute globe origo in world space
+        const matrix = calculateGlobeMatrix(this._tr, this._worldSize);
```

Comment:
This could be replaced with the cached matrix?

Reply:
https://github.com/mapbox/mapbox-gl-js/pull/11329/commits/6fc7b2db0e69aa14194f75c4c9a296bf7baed891